src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -219,98 +186,6 @@ public override void Dispose()
         private void OnDisconnected(JsonRpcDisconnectedEventArgs e)
             => Dispose();
 
-        #region Global Operation Notifications
-
-        private void RegisterGlobalOperationNotifications()
-        {
-            var globalOperationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
-            if (globalOperationService != null)
-            {
-                globalOperationService.Started += OnGlobalOperationStarted;
-                globalOperationService.Stopped += OnGlobalOperationStopped;
-            }
-        }
-
-        private void UnregisterGlobalOperationNotifications()
-        {
-            var globalOperationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
-            if (globalOperationService != null)
-            {
-                globalOperationService.Started -= OnGlobalOperationStarted;
-                globalOperationService.Stopped -= OnGlobalOperationStopped;
-            }
-
-            Task localTask;
-            lock (_globalNotificationsGate)
-            {
-                // Unilaterally transition us to the finished state.  Once we're finished
-                // we cannot start or stop anymore.
-                _globalNotificationsTask = _globalNotificationsTask.ContinueWith(
-                    _ => GlobalNotificationState.Finished, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default);
-                localTask = _globalNotificationsTask;
-            }
-
-            // Have to wait for all the notifications to make it to the OOP side so we keep
-            // it in a consistent state.  Also, if we don't do this, our _rpc object will
-            // get disposed while we're remoting over the messages to the oop side.
-            localTask.Wait();
-        }
-
-        private void OnGlobalOperationStarted(object sender, EventArgs e)
-        {
-            lock (_globalNotificationsGate)
-            {
-                _globalNotificationsTask = _globalNotificationsTask.SafeContinueWithFromAsync(
-                    continuation, _shutdownCancellationTokenSource.Token, TaskContinuationOptions.None, TaskScheduler.Default);
-            }
-
-            async Task<GlobalNotificationState> continuation(Task<GlobalNotificationState> previousTask)
-            {
-                // Can only transition from NotStarted->Started.  If we hear about
-                // anything else, do nothing.
-                if (previousTask.Result != GlobalNotificationState.NotStarted)
-                {
-                    return previousTask.Result;
-                }
-
-                await _endPoint.InvokeAsync(
-                    nameof(IRemoteHostService.OnGlobalOperationStarted),
-                    new object[] { "" },
```

Comment:
previously we sent empty string, now we send nothing, is that ok?

Reply:
Yes, there probably used to be some bug in JSON-RPC. I tested this works.

---
In reply to: [419027340](https://github.com/dotnet/roslyn/pull/43928#discussion_r419027340) [](ancestors = 419027340)